discussions/views.py

```diff
@@ -0,0 +1,51 @@
+"""APIs for discussions"""
+from django.conf import settings
+from django.shortcuts import redirect
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from discussions.utils import get_token_for_user
+
+
+def _set_jwt_cookie(response, token):
+    """
+    Set the token on the response
+
+    Args:
+        response (django.http.Response): the response to set the cookie on
+        token (str): the JWT token
+    """
+    response.set_cookie(
+        settings.OPEN_DISCUSSIONS_COOKIE_NAME,
+        token,
+        domain=settings.OPEN_DISCUSSIONS_COOKIE_DOMAIN,
+        httponly=True,
+        max_age=settings.OPEN_DISCUSSIONS_COOKIE_EXPIRES_DELTA
+    )
+
+
+@api_view()
+def discussions_token(request):
+    """
+    API view for setting discussions JWT token
+    """
+    token = get_token_for_user(request.user)
+    response = Response({
+        'token': token is not None
+    })
+    if token is not None:
+        _set_jwt_cookie(response, token)
+    return response
+
+
+def discussions_redirect(request):
+    """
+    View for setting discussions JWT token and redirecting to discussions
+    """
+    token = get_token_for_user(request.user)
+    if token is not None:
+        response = redirect(settings.OPEN_DISCUSSIONS_REDIRECT_URL)
+        _set_jwt_cookie(response, token)
+    else:
+        response = redirect('/' if request.user.is_anonymous() else '/dashboard')
```

Comment:
What is the use case for redirecting here? I feel like raising an exception would be more appropriate. If the user is anonymous we should use `@login_required` to protect against access. The user should have the correct `DiscussionUser` set up, at least in production since they will have saved their profile before going here (or we do this manually once for existing users).

Reply:
The creation of `DiscussionUser` for new users is deferred to a celery task, so while the record *should* exist, it's not guaranteed to. I can raise an exception though if you feel that's a better approach.